openproblems/tasks/spatial_decomposition/methods/nmfreg.py

```diff
@@ -0,0 +1,97 @@
+from ....tools.decorators import method
+from ....tools.normalize import log_cpm
+from ....tools.utils import check_version
+from sklearn.decomposition import NMF
+from sklearn.preprocessing import StandardScaler
+
+import collections
+import numpy as np
+import pandas as pd
+import scipy.optimize
+import scipy.stats
+
+
+@method(
+    method_name="NMF-reg",
+    paper_name="Slide-seq: A scalable technology for measuring genome-wide expression at high spatial resolution",
+    paper_url="https://science.sciencemag.org/content/363/6434/1463",
+    paper_year=2009,
+    code_url="N/A",
+    code_version=check_version("NMF-reg"),
+)
+def nmfreg(adata):
+    n_types = adata.obsm["proportions_true"].size[1]
+    adata_sc = adata.uns["sc_reference"].copy()
+    factors = 30
+    projectiob_type = "l2"
+
+    # Learn from reference
+    X = adata_sc.X.toarray()
+    X_norm = X / X.sum(1)[:, np.newaxis]
+    X_scaled = StandardScaler(with_mean=False).fit_transform(X_norm)
+
+    model = NMF(n_components=factors, init="random", random_state=17)
```

Comment:
that's a very good point re random seed, is there any best practice in the repo? will ask